### PR TITLE
Improved artsiomkorzun solution

### PR DESCRIPTION
**Pull request by artsiomkorzun**
Improvements:
* Reworked parallel computations using threads
* Saving hash and using it
* Unrolled parsing
* Use GraalVM

### CODE REVIEW

**Comment by gunnarmorling** [comment]
@artsiomkorzun, could you rebase this to current main and squash everything into one commit?

File: calculate_average_artsiomkorzun.sh [hunk 1]
```diff
@@ -17,4 +17,5 @@
 
 
 JAVA_OPTS="-XX:+UseParallelGC"
+sdk use java 21.0.1-graal
```
[the remark this answers is not in the document]
Comment (artsiomkorzun) [reply calculate_average_artsiomkorzun.sh]: Thanks. Changed

**Comment by artsiomkorzun** [comment]
> @artsiomkorzun, could you rebase this to current main and squash everything into one commit?

Done

**Comment by gunnarmorling** [comment]
00:10.589, nice!

**Pull request merged**